cachito/web/models.py

```diff
@@ -172,18 +172,21 @@ def to_json(self):
 
         env_vars_json = OrderedDict(env_var.to_json() for env_var in self.environment_variables)
         rv = {
-            'dependencies': [dep.to_json() for dep in self.dependencies],
             'id': self.id,
             'repo': self.repo,
             'ref': self.ref,
             'pkg_managers': pkg_managers,
-            'state_history': states,
             'user': user,
             'environment_variables': env_vars_json,
             'flags': [flag.to_json() for flag in self.flags if flag.active],
         }
         # Show the latest state information in the first level of the JSON
         rv.update(latest_state)
+        if verbose:
+            rv.update({'state_history': states})
+            rv.update({'dependencies': [dep.to_json() for dep in self.dependencies]})
+        else:
+            rv.update({'dependencies': len(self.dependencies)})
```

Comment:
This will pull down all the dependencies from the database and then provide the length of the list. This is inefficient and will make the REST API slower. How about using SQL to get the count?

For instance:
```python
import sqlalchemy

req_dep_cols = request_dependency_table.c
num_deps = db.session.query(sqlalchemy.func.count(req_dep_cols.dependency_id)).filter(req_dep_cols.request_id == self.id).scalar()
```

You could put the code snippet in it's own property method called `number_of_deps` for readability.